app/models/user_block.rb

```diff
@@ -27,6 +27,12 @@ def active?
   end
 
   ##
+  # returns true if the block is a "zero hour" block
+  def zero_hour?
+    needs_view && (ends_at.to_i == updated_at.to_i)
```

Comment:
"Is needs_view not what you really want to check?" I believe that is essentially always set by the DWG, including longer blocks. It isn't an indicator of  the block being a zero hour one,  I suppose testing if ends_at.to_i - updated_at.to_i is smaller than a suitable value should be robust enough.